src/pages/ReportParticipantsPage.tsx

```diff
@@ -1,135 +1,394 @@
+import Text from '@components/Text';
 import useLocalize from '@hooks/useLocalize';
+import useStyleUtils from '@hooks/useStyleUtils';
 import useThemeStyles from '@hooks/useThemeStyles';
-import localeCompare from '@libs/LocaleCompare';
-import * as LocalePhoneNumber from '@libs/LocalePhoneNumber';
-import type * as Localize from '@libs/Localize';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import * as Report from '@libs/actions/Report';
 import Navigation from '@libs/Navigation/Navigation';
 import * as PersonalDetailsUtils from '@libs/PersonalDetailsUtils';
 import * as ReportUtils from '@libs/ReportUtils';
-import type {OptionData} from '@libs/ReportUtils';
 import * as UserUtils from '@libs/UserUtils';
 import CONST from '@src/CONST';
-import type {TranslationPaths} from '@src/languages/types';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
-import type {PersonalDetailsList, Report} from '@src/types/onyx';
+import type {PersonalDetailsList, Session} from '@src/types/onyx';
+import type {Errors} from '@src/types/onyx/OnyxCommon';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import Log from '@libs/Log';
 import type {WithReportOrNotFoundProps} from './home/report/withReportOrNotFound';
 import withReportOrNotFound from './home/report/withReportOrNotFound';
 
 type ReportParticipantsPageOnyxProps = {
     /** Personal details of all the users */
     personalDetails: OnyxEntry<PersonalDetailsList>;
+
+    /** Session info for the currently logged in user. */
+    session: OnyxEntry<Session>;
 };
 
 type ReportParticipantsPageProps = ReportParticipantsPageOnyxProps & WithReportOrNotFoundProps;
 
-/**
- * Returns all the participants in the active report
- *
- * @param report The active report object
- * @param personalDetails The personal details of the users
- * @param translate The localize
- */
-const getAllParticipants = (
-    report: OnyxEntry<Report>,
-    personalDetails: OnyxEntry<PersonalDetailsList>,
-    translate: <TKey extends TranslationPaths>(phraseKey: TKey, ...phraseParameters: Localize.PhraseParameters<Localize.Phrase<TKey>>) => string,
-): OptionData[] =>
-    ReportUtils.getVisibleMemberIDs(report)
-        .map((accountID, index) => {
-            const userPersonalDetail = personalDetails?.[accountID];
-            const userLogin =
-                !!userPersonalDetail?.login && !CONST.RESTRICTED_ACCOUNT_IDS.includes(accountID) ? LocalePhoneNumber.formatPhoneNumber(userPersonalDetail.login) : translate('common.hidden');
-            const displayName = PersonalDetailsUtils.getDisplayNameOrDefault(userPersonalDetail);
-
-            const pendingChatMember = report?.pendingChatMembers?.findLast((member) => member.accountID === accountID.toString());
-            return {
-                alternateText: userLogin,
-                pendingAction: pendingChatMember?.pendingAction,
```

Comment:
Is this no longer needed?

Reply:
Looking into it now. I think this was mainly used for the `InviteToRoom` flow. But these flows ideally should have the same behavior with regards to the "pending" users (added while offline).

Reply:
Alright, so seems like the history for this comes from [this issue](https://github.com/Expensify/App/issues/39985). Sorry, this new detail caught me by surprise as it was implemented somewhat recently. 

Prior to Group Chats I _believe_ these are the types of report that can have "pending members"

- "workspace room" a.k.a. User Created Room. Here, "pendingChatMember" can only exist if we invite or remove someone from a "room" (for now, it seems we are using a different view, but I would expect us to retire that view in the future and just use the `ReportParticipantsPage` for all the chats)
- "announce room" when we add or remove a member to a workspace they could become a "pendingChatMember"

So, I think we definitely _do_ need this here because viewing the members of an "announce" room or workspace chat or whatever is possible via this flow.

Reply:
I'll look at adding this back in so we don't regress that behavior. It's also crashing with these repro steps:

- Invite someone to a workspace
- Navigate to `#announce` channel for workspace
- Tap report header avatar
- Tap "Members"
- 💥

Reply:
Ah, I am also realizing that these types of members could potentially be in `report.visibleChatMemberAccountIDs`, but might not be in `report.participants` 🙃 

And now that we are computing the `visibleChatMemberAccountIDs` by looking at the participants some optimistic changes will not work the same. So, yes, some things need to change here and also we should be using the "visible participants" for any chats besides Group.